src/SSymmetricCompact.jl

```diff
@@ -0,0 +1,239 @@
+struct SSymmetricCompact{N, T, L} <: StaticMatrix{N, N, T}
+    lowertriangle::SVector{L, T}
+
+    @inline function SSymmetricCompact{N, T, L}(lowertriangle::SVector{L}) where {N, T, L}
+        check_symmetric_parameters(Val(N), Val(L))
+        new{N, T, L}(lowertriangle)
+    end
+end
+
+@generated function check_symmetric_parameters(::Val{N}, ::Val{L}) where {N, L}
+    if 2 * L != N * (N + 1)
+        return :(throw(ArgumentError("Size mismatch in SSymmetricCompact parameters. Got dimension $N and length $L.")))
+    end
+    :(nothing)
+end
+
+Base.@pure triangularnumber(N::Int) = div(N * (N + 1), 2)
+Base.@pure triangularroot(L::Int) = div(isqrt(8 * L + 1) - 1, 2) # from quadratic formula
+
+lowertriangletype(::Type{SSymmetricCompact{N, T, L}}) where {N, T, L} = SVector{L, T}
+lowertriangletype(::Type{SSymmetricCompact{N, T}}) where {N, T} = SVector{triangularnumber(N), T}
+lowertriangletype(::Type{SSymmetricCompact{N}}) where {N} = SVector{triangularnumber(N)}
+
+@inline (::Type{SSymmetricCompact{N, T}})(lowertriangle::SVector{L}) where {N, T, L} = SSymmetricCompact{N, T, L}(lowertriangle)
+@inline (::Type{SSymmetricCompact{N}})(lowertriangle::SVector{L, T}) where {N, T, L} = SSymmetricCompact{N, T, L}(lowertriangle)
+
+@inline function SSymmetricCompact(lowertriangle::SVector{L, T}) where {T, L}
+    N = triangularroot(L)
+    SSymmetricCompact{N, T, L}(lowertriangle)
+end
+
+@generated function (::Type{SSymmetricCompact{N, T, L}})(a::Tuple) where {N, T, L}
```

Comment:
What size should `a` have? Because of your formula for `index`, it seems it's of length N^2? This needs to be checked.

Reply:
This is now documented in the `SSymmetricCompact` docstring.